### PR TITLE
feat: Flip sampling defaults to false

### PR DESCRIPTION
**Pull request by lynnagara**
This only affects open source users as "projects:sample-events"
should already always be false for all SaaS users on sentry.io.

We should follow up by removing all of the associated sampling code.

### CODE REVIEW

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -881,7 +881,7 @@ def create_partitioned_queues(name):
     # Enable functionality for rate-limiting events on projects.
     "projects:rate-limits": True,
     # Enable functionality for sampling of events on projects.
-    "projects:sample-events": True,
+    "projects:sample-events": False,
```
Comment (markstory) [inline src/sentry/conf/server.py]: Do we need to retain this feature switch if its not used on saas?

Comment (lynnagara) [reply src/sentry/conf/server.py]: It's still being referenced right now, will remove it together with the code

**Comment by fpacifici** [comment]
Technically someone using the open source version may still be relying on this. Do we need to communicate it before removing the feature ?

**Comment by lynnagara** [comment]
@BYK Do you have any thoughts on this? For any open source users who are running Snuba / 10.x sampling probably hasn't worked as expected for quite a while.

**Comment by BYK** [comment]
Thanks for the heads up @lynnagara! We've turned this off for v10 onpremise per @jan-auer's suggestion earlier: https://github.com/getsentry/onpremise/pull/220/files#diff-66f7c6c9e9f860724fb428123754dbaaR183

**Comment by lynnagara** [comment]
It sounds like we don't need to communicate anything extra then. Since the feature is already kind of broken and we're removing it soon we could probably go ahead and take it out of the docs then to avoid any confusion. https://github.com/getsentry/sentry-docs/pull/1216

**Review by fpacifici — APPROVED** [review]

**Pull request merged**